packages/paste-core/components/code-block/src/CodeBlockTabList.tsx

```diff
@@ -17,12 +77,67 @@ export interface CodeBlockTabListProps extends Omit<TabListProps, "aria-label">
 
 export const CodeBlockTabList = React.forwardRef<HTMLDivElement, CodeBlockTabListProps>(
   ({ children, element = "CODE_BLOCK_TAB_LIST", ...props }, ref) => {
+    const tab = React.useContext(TabsContext);
+    const theme = useTheme();
+    //  ref to the scrollable element
+    const scrollableRef = React.useRef<HTMLDivElement>(null);
+    const [scrollShadow, setScrollShadow] = React.useState<"none" | "left" | "right" | "both">("none");
+
+    // Function to handle scroll event
+    const handleScroll = (): void => {
+      if (scrollableRef.current) {
+        // No scrollbar, so no shadow
+        if (scrollableRef.current.clientWidth === scrollableRef.current.scrollWidth) {
+          setScrollShadow("none");
+        }
+        // We're positioned on the left most side, so only show right shadow
+        else if (scrollableRef.current.scrollLeft === 0) {
+          setScrollShadow("right");
+        } else if (
+          // We're positioned on the right most side, so only show left shadow
+          scrollableRef.current.scrollLeft + scrollableRef.current.clientWidth ===
+          scrollableRef.current.scrollWidth
+        ) {
+          setScrollShadow("left");
+        } else {
+          // Show both shadows
+          setScrollShadow("both");
+        }
+      }
+    };
+
     return (
-      <Box paddingX="space70">
-        <TabList {...props} aria-label="label" ref={ref} element={element}>
-          {children}
-        </TabList>
-      </Box>
+      <TabPrimitiveList {...(tab as any)} as={Box} {...props} element={element} ref={ref}>
+        <Box
+          element={`${element}_CHILD_WRAPPER`}
+          marginLeft="space70"
```

Comment:
changed to margin instead of padding as the absolute position of the left div for left shadow would also need to use the design token. Now it can just be aligned to start of container

Reply:
```suggestion
          marginX="space70"
```

Also not seeing the scrollbar

![Screenshot 2024-09-12 at 8 33 27 PM](https://github.com/user-attachments/assets/cb039d1c-0470-482f-95f5-1e36d11b968d)

Reply:
It shouldn't be x as the margin is only so that the wrapper aligns with the header. ther tabs should extend all teh way to the end of the component. marginX would not allow that 
![image](https://github.com/user-attachments/assets/3ff66356-749e-4bb1-b3a8-c17bc96895a8)

Reply:
can we confirm with design that it should be marginLeft and not marginX? looks kinda jank to have a left margin but not a right margin on scroll (like in the screenshot I posted)

cc @twilio-labs/design-systems-pd